recipes-bsp/device-tree/files/tepzu.dts

```diff
@@ -0,0 +1,370 @@
+	snps,dis_u3_susphy_quirk;
+	/delete-property/ phy-names;
+	/delete-property/ phys;
+	/delete-property/ snps,usb3_lpm_capable;
+	vbus-supply = <&reg_usb0_vbus>;
+	connector {
+		compatible = "usb-b-connector";
+		label = "micro-USB-otg";
+		type = "micro";
+	};
+};
+
+&usb1 {
+	status = "okay";
+	/*
+	 * Using Xilinx' borked implementation to add a reset for the HUB,
+	 * otherwise SuperSpeed doesn't work.
+	 */
+	reset-gpios = <&tep_gpioex10 5 GPIO_ACTIVE_LOW>;
+};
+
+&dwc3_1 {
+	status = "okay";
+	dr_mode = "host";
+	/* refclk0 from clock synth */
+	assigned-clocks = <&si5345 0 3>;
+	assigned-clock-rates = <100000000>;
+	phy-names = "usb3-phy";
+	phys = <&psgtr 3 PHY_TYPE_USB3 1 0>;
+	maximum-speed = "super-speed";
+	snps,usb3_lpm_capable;
+};
+
+&pcie {
+	status = "okay";
+	phys = <&psgtr 0 PHY_TYPE_PCIE 0 0>;
+	/* refclk0 from clock synth */
+	assigned-clocks      = <&si5345 0 3>;
+	assigned-clock-rates =   <100000000>;
+};
+
+/* Displayport - HP and AUX signals through EMIO, enabled when PL probes */
+&zynqmp_dpsub {
+	phy-names = "dp-phy0";
+	phys = <&psgtr 1 PHY_TYPE_DP 0 3>;
+	xlnx,max-lanes = <1>;
+	assigned-clocks = <&si5345 0 8>;
+	assigned-clock-rates = <27000000>;
+	assigned-clock-parents = <&si5345 1 1>;
+};
+&zynqmp_dpdma {
+	status = "okay";
+};
+
+&si5345 {
+	assigned-clock-parents = <0>, <0>, <0>, <0>, <0>,
+				<&si5345 1 0>, /* out 0 */
+				<&si5345 1 0>,
+				<&si5345 1 0>,
+				<&si5345 1 0>,
+				<&si5345 1 0>,
+				<&si5345 1 0>, /* out 5 */
+				<&si5345 1 0>,
+				<&si5345 1 0>,
+				<&si5345 1 1>,
+				<&si5345 1 0>;
+	assigned-clock-rates =	<1000000000>, /* synth 0 */
+				< 216000000>, /* Set synth 1 to 8 * 27 MHz */
+				<0>,
+				<0>,
+				<0>,
+				<125000000>, /* out 0 */
+				<125000000>,
+				<125000000>,
+				<100000000>,
+				<125000000>,
+				< 25000000>, /* out 5 (ethernet) */
+				<100000000>,
+				<250000000>,
+				< 27000000>, /* out 8 (PS refclk3 Displayport) */
+				<100000000>;
+
+	out@0 {
+		always-on;
+	};
+	out@1 {
+		always-on;
+	};
+	out@4 {
+		/* 125MHz Clock for MGT bank 130 (firefly) */
+		always-on;
+	};
+	out@8 {
+		/* Displayport clock needs to generate 27MHz */
+		silabs,pll-master;
+	};
+};
+
+/* Carrier provides VBAT for RTC */
+&rtc {
+	status = "okay";
+};
+
+&gpio {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio_default>;
+};
+
+&pinctrl0 {
+	status = "okay";
+
+	pinctrl_sdhci1_default: sdhci1-sd-default {
+		/* SD using pins 46 .. 51 in 4-bit mode */
+		mux {
+			groups = "sdio1_2_grp";
+			function = "sdio1";
+		};
+		conf {
+			groups = "sdio1_2_grp";
+			slew-rate = <SLEW_RATE_FAST>;
+			power-source = <IO_STANDARD_LVCMOS18>;
+			bias-disable;
+		};
+
+		/* Card detect on pin 45 */
+		mux-cd {
+			groups = "gpio0_45_grp";
+			function = "gpio0";
+		};
+		conf-cd {
+			groups = "gpio0_45_grp";
+			bias-high-impedance;
+			slew-rate = <SLEW_RATE_SLOW>;
+			power-source = <IO_STANDARD_LVCMOS18>;
+			bias-pull-up;
+		};
+	};
+
+	pinctrl_gpio_default: gpio-default {
```

Comment:
When is this pinctrl_.. setup in the device tree required?
Is it to define the IO standards for MIO pins, not handled by Vivado?

Reply:
Basically, yes, allows to override pinmux values without having to build a new bootloader.

There are more applications for this, like being able to disable pull-ups while the device is sleeping, or to be able to recover an I2C bus by switching the associated pins to GPIO mode.